storage/2020-08-04/file/shares/api.go

```diff
@@ -2,23 +2,20 @@ package shares
 
 import (
 	"context"
-
-	"github.com/Azure/go-autorest/autorest"
 )
 
 type StorageShare interface {
-	SetACL(ctx context.Context, accountName, shareName string, acls []SignedIdentifier) (result autorest.Response, err error)
-	GetSnapshot(ctx context.Context, accountName, shareName, snapshotShare string) (result GetSnapshotPropertiesResult, err error)
-	GetStats(ctx context.Context, accountName, shareName string) (result GetStatsResult, err error)
-	GetACL(ctx context.Context, accountName, shareName string) (result GetACLResult, err error)
-	SetMetaData(ctx context.Context, accountName, shareName string, metaData map[string]string) (result autorest.Response, err error)
-	GetMetaData(ctx context.Context, accountName, shareName string) (result GetMetaDataResult, err error)
-	SetProperties(ctx context.Context, accountName, shareName string, properties ShareProperties) (result autorest.Response, err error)
-	DeleteSnapshot(ctx context.Context, accountName, shareName string, shareSnapshot string) (result autorest.Response, err error)
-	CreateSnapshot(ctx context.Context, accountName, shareName string, input CreateSnapshotInput) (result CreateSnapshotResult, err error)
-	GetResourceID(accountName, shareName string) string
+	SetACL(ctx context.Context, shareName string, input SetAclInput) (resp setAclResponse, err error)
```

Comment:
(as above) we don't need to name the return args here:

```suggestion
	SetACL(ctx context.Context, shareName string, input SetAclInput) (setAclResponse, error)
```

also `setAclResponse` should be public